src/machines/onboardingMachine.tsx

```diff
@@ -65,146 +117,22 @@ const createOnboardingMachine = (opts?: CreateOnboardingMachineOpts) => {
-        on: {
-          [OnboardingMachineEvents.SET_PERSONAL_DATA]: {
-            actions: assign({personalData: (_ctx: OnboardingMachineContext, e: PersonalDataEvent) => e.data}),
-          },
-          [OnboardingMachineEvents.NEXT]: {
-            cond: OnboardingMachineGuards.onboardingPersonalDataGuard,
-            target: OnboardingMachineStates.enterPin,
-          },
-          [OnboardingMachineEvents.PREVIOUS]: {target: OnboardingMachineStates.acceptAgreement},
-        },
-      },
-      [OnboardingMachineStates.enterPin]: {
-        on: {
-          [OnboardingMachineEvents.SET_PIN]: {
-            actions: assign({pinCode: (_ctx: OnboardingMachineContext, e: PinSetEvent) => e.data}),
-          },
-          [OnboardingMachineEvents.NEXT]: {
-            cond: OnboardingMachineGuards.onboardingPinCodeSetGuard,
-            target: OnboardingMachineStates.verifyPin,
-          },
-          [OnboardingMachineEvents.PREVIOUS]: {
-            target: OnboardingMachineStates.enterPersonalDetails,
-          },
-        },
-      },
-      [OnboardingMachineStates.verifyPin]: {
-        on: {
-          [OnboardingMachineEvents.NEXT]: {
-            cond: OnboardingMachineGuards.onboardingPinCodeVerifyGuard,
-            target: OnboardingMachineStates.verifyPersonalDetails,
-          },
-          [OnboardingMachineEvents.PREVIOUS]: {
-            target: OnboardingMachineStates.enterPin,
-          },
-        },
-      },
-      [OnboardingMachineStates.verifyPersonalDetails]: {
-        on: {
-          [OnboardingMachineEvents.NEXT]: {
-            target: OnboardingMachineStates.setupWallet,
-          },
-          [OnboardingMachineEvents.PREVIOUS]: {
-            target: OnboardingMachineStates.enterPin, // We are going back to pin entry and then verify
-          },
-        },
-      },
-      [OnboardingMachineStates.setupWallet]: {
-        invoke: {
-          id: OnboardingMachineStates.setupWallet,
-          src: OnboardingMachineStates.setupWallet,
-          onDone: {
-            target: OnboardingMachineStates.finishOnboarding,
-          },
-          // todo: On Error
-        },
-      },
-      [OnboardingMachineStates.declineOnboarding]: {
-        id: OnboardingMachineStates.declineOnboarding,
-        always: OnboardingMachineStates.showIntro,
-        entry: assign({
-          ...initialContext,
-        }),
-        // Since we are not allowed to exit an app by Apple/Google, we go back to the onboarding state when the user declines
-      },
-      [OnboardingMachineStates.finishOnboarding]: {
-        type: 'final',
-        id: OnboardingMachineStates.finishOnboarding,
-        entry: assign({
-          pinCode: '',
-          personalData: undefined,
-          credentialData: undefined,
-          privacyPolicyAccepted: false,
-          termsConditionsAccepted: false,
-        }),
-      },
-    },
+    initial: OnboardingMachineStateType.showIntro,
```

Comment:
also it seems to miss step 3 and onwards. i am not sure if want to layout the whole machine or just a part of it

Reply:
As i mentioned in the [description](https://github.com/Sphereon-Opensource/mobile-wallet/pull/204#issue-2451020063), these were removed as they are not used or should be defined in the current state of the onboarding machine. They should be re-introduced when we implement the relevant screens.

Reply:
Merveille will be adding the new states as he implements step 3 (import personal data)

Reply:
but you are using events, and it would be nice if we turn the conditions into actual guards like we do within the current machines. i am also fine to say ok, we can do that later, but then we need to make sure that happens

Reply:
👍🏻

Reply:
added two guards